modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -123,11 +123,13 @@ public void setRevisionCallback(OnRevisionAppliedCallback revisionCallback) {
 
     /**
      * Notifies registered watches about an update event.
+     *
+     * <p>This method is not thread-safe and must be performed under an exclusive lock in concurrent scenarios.
      */
-    public void notifyWatches(List<Entry> updatedEntries, HybridTimestamp time) {
+    public CompletableFuture<Void> notifyWatches(List<Entry> updatedEntries, HybridTimestamp time) {
```

Comment:
It's a little strange for me to notice that the method returns the future but does not wait for it in the core code.

Reply:
I don't know how to address this comment =(

Reply:
We need to fix this.

Reply:
How? What's wrong with the current approach?